File: go/equal-row-and-column-pairs.go
func equalPairs(grid [][]int) int {
	rows := make(map[string]int)
	columns := make(map[string]int)
	n := len(grid)
	for i := 0; i < n; i++ {
		row := ""
		column := ""
		for j := 0; j < n; j++ {
			row += strconv.Itoa(grid[i][j]) + "_"
			column += strconv.Itoa(grid[j][i]) + "_"
		}
		rows[row] += 1
		columns[column] += 1
	}
	pairs := 0
	for row := range rows {
		pairs += rows[row] * columns[row]
	}
	return pairs
}

File: python/equal-row-and-column-pairs.py
class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:
        rows = defaultdict(lambda: 0)
        columns = defaultdict(lambda: 0)
        for i in range(len(grid)):
            row = "_".join(map(str, grid[i]))
            rows[row] += 1
            column = "_".join([str(elements[i]) for elements in grid])
            columns[column] += 1
        pairs = 0
        for row in rows:
            pairs += rows[row] * columns[row]
        return pairs
